internal/grpc/services/gateway/ocmshareprovider.go

```diff
@@ -219,6 +221,70 @@ func (s *svc) UpdateReceivedOCMShare(ctx context.Context, req *ocm.UpdateReceive
 		}, nil
 	}
 
+	getShareReq := &ocm.GetReceivedOCMShareRequest{Ref: req.Ref}
+	getShareRes, err := s.GetReceivedOCMShare(ctx, getShareReq)
+	if err != nil {
+		log.Err(err).Msg("gateway: error calling GetReceivedShare")
+		return &ocm.UpdateReceivedOCMShareResponse{
+			Status: &rpc.Status{
+				Code: rpc.Code_CODE_INTERNAL,
+			},
+		}, nil
+	}
+
+	if getShareRes.Status.Code != rpc.Code_CODE_OK {
+		log.Error().Msg("gateway: error calling GetReceivedShare")
+		return &ocm.UpdateReceivedOCMShareResponse{
+			Status: &rpc.Status{
+				Code: rpc.Code_CODE_INTERNAL,
+			},
+		}, nil
+	}
+
+	share := getShareRes.Share
+	if share == nil {
+		panic("gateway: error updating a received share: the share is nil")
+	}
+
+	if share.GetShare().ShareType == ocm.Share_SHARE_TYPE_TRANSFER {
+		srcRemote := share.GetShare().GetOwner().GetIdp()
+		// TODO do we actually know for sure the home path of the src reva instance?
```

Comment:
Take a look at how we do it for webdav access https://github.com/cs3org/reva/blob/master/internal/grpc/services/gateway/webdavstorageprovider.go#L215

Reply:
@ishank011 : I need some help here, I can't figure out how to retrieve the target uri in `datatx.CreateTransfer` from the `cs3.storage.provider.v1beta1.Reference`. I know it's stored when de OCM reference is created but ...